crates/ra_cli/src/main.rs

```diff
@@ -13,93 +14,123 @@ type Result<T> = std::result::Result<T, Box<dyn Error + Send + Sync>>;
+                )?;
+            }
         }
-        ("analysis-bench", Some(matches)) => {
-            let verbose = matches.is_present("verbose");
-            let path = matches.value_of("path").unwrap_or("");
-            let op = if let Some(path) = matches.value_of("highlight") {
-                analysis_bench::Op::Highlight { path: path.into() }
-            } else if let Some(path_line_col) = matches.value_of("complete") {
-                let (path_line, column) = rsplit_at_char(path_line_col, ':')?;
-                let (path, line) = rsplit_at_char(path_line, ':')?;
-                analysis_bench::Op::Complete {
-                    path: path.into(),
-                    line: line.parse()?,
-                    column: column.parse()?,
-                }
+        "analysis-bench" => {
+            if matches.contains(["-h", "--help"]) {
+                help::print_analysis_bench_help();
+                return Ok(());
             } else {
-                panic!("either --highlight or --complete must be set")
-            };
-            analysis_bench::run(verbose, path.as_ref(), op)?;
+                let verbose = matches.contains(["-v", "--verbose"]);
+                let path = matches.value_from_str("--path")?.unwrap_or("".to_string());
+                let highlight_path = matches.value_from_str("--highlight")?;
+                let complete_path = matches.value_from_str("--complete")?;
+                if highlight_path.is_some() && complete_path.is_some() {
+                    panic!("either --highlight or --complete must be set, not both")
+                }
+                let op = if let Some(path) = highlight_path {
+                    let path: String = path;
+                    analysis_bench::Op::Highlight { path: path.into() }
+                } else if let Some(path_line_col) = complete_path {
+                    let path_line_col: String = path_line_col;
+                    let (path_line, column) = rsplit_at_char(path_line_col.as_str(), ':')?;
+                    let (path, line) = rsplit_at_char(path_line, ':')?;
+                    analysis_bench::Op::Complete {
+                        path: path.into(),
+                        line: line.parse()?,
+                        column: column.parse()?,
+                    }
+                } else {
+                    panic!("either --highlight or --complete must be set")
+                };
+                matches.finish().or_else(handle_extra_flags)?;
+                analysis_bench::run(verbose, path.as_ref(), op)?;
+            }
         }
-        _ => unreachable!(),
+        _ => help::print_global_help(),
     }
     Ok(())
 }
 
+fn handle_extra_flags(e: pico_args::Error) -> Result<()> {
+    if let pico_args::Error::UnusedArgsLeft(flags) = e {
+        let mut invalid_flags = String::new();
+        for flag in flags {
+            write!(&mut invalid_flags, "{}, ", flag).expect("Error on write");
```

Comment:
let's `?`

Reply:
You mean no need for `handle_eytra_flags`?

Reply:
I mean replace `.expect` with `?`